lib/vsc/install/headers.py

```diff
@@ -180,12 +180,39 @@ def check_header(filename, script=False, write=False):
     Return if header is different from expected or not
     """
 
-    header, shebang = get_header(filename, script=script)
+    header, orig_shebang = get_header(filename, script=script)
     header_end_pos = len(header)
+
     changed = False
-    if shebang is not None:
+
+    shebang = None
+    if script:
+        # scripts must have an appropriate shebang
+        shebang = orig_shebang
+        file_ext = os.path.splitext(filename)[1]
+        log.info("Shebang found in %s (ext: %s): %s", filename, file_ext, shebang)
+
+        if file_ext == '.py':
+            if shebang != SHEBANG_ENV_PYTHON:
+                log.info("Wrong shebang for Python script %s: found '%s', should be '%s'",
+                        filename, shebang, SHEBANG_ENV_PYTHON)
+                shebang = SHEBANG_ENV_PYTHON
+
+        elif file_ext in ['.sh', '']:
+            if shebang != SHEBANG_BIN_BASH:
+                log.info("Wrong shebang for shell script %s: found '%s', should be '%s'",
+                         filename, shebang, SHEBANG_BIN_BASH)
+                shebang = SHEBANG_BIN_BASH
+
+        else:
+            log.warn("Don't know expected shebang based on extension '%s' for script '%s', assuming it's OK...",
+                     file_ext, filename)
+
+        changed = shebang != orig_shebang
+
+    if orig_shebang is not None:
         # original position
-        header_end_pos += 1 + len(shebang)  # 1 is from splitted newline
+        header_end_pos += 1 + len(orig_shebang)  # 1 is from splitted newline
 
         if 'python' in shebang and shebang != SHEBANG_ENV_PYTHON:
```

Comment:
@JensTimmerman It only changed the shebang if i) if there was a shebang (see also `shebang is not None` above), ii) if it has `python` in it and it wasn't the expected `#!/usr/bin/env python`

Reply:
that's only the code that does the checking and auto fixing of headers.

I'm saying that vsc-install adds the shebang sat install time, not at test time

https://github.com/hpcugent/vsc-install/blob/be3a2b0261cd497b4dd9e286242f94becd46cc83/lib/vsc/install/shared_setup.py#L621

Reply:
but currently only for python scripts, not for bash scripts. might need to be extended for that.

Reply:
I seem to be confused, that line also only does the replacement of env to non env
there used to be code around somewhere that added shebangs, but I can't seem to find it

Reply:
If it was there, https://github.ugent.be/hpcugent/vsc-filesystems-quota/pull/40 wouldn't have happened.

Also, there was no test that checked whether Python scripts with missing shebang was being corrected, now there is.

Reply:
The python scripts had no shebang line after installation (and did not have any in the repo) when https://github.ugent.be/hpcugent/vsc-filesystems-quota/pull/39 was merged.